src/coreclr/gc/gc.cpp

```diff
@@ -45277,6 +45277,15 @@ HRESULT GCHeap::Initialize()
             // If no hard_limit is configured the reservation size is max of 256gb or 2x physical limit
             gc_heap::regions_range = max(((size_t)256 * 1024 * 1024 * 1024), (size_t)(2 * gc_heap::total_physical_mem));
         }
+        size_t vmLimit = GCToOSInterface::GetVirtualMemoryLimit();
+        if (vmLimit != (size_t)-1)
+        {
+            vmLimit = vmLimit * (size_t)75 / (size_t)100;
```

Comment:
@cshung I was going to make this ratio configurable, but I have noticed that similar limit above for the physical memory has also fixed ratio of 75%. So I am not sure anymore whether to make this configurable or not.